netmiko/oneaccess/oneaccess_oneos.py

```diff
@@ -16,14 +16,22 @@ def session_preparation(self) -> None:
         """Prepare connection - disable paging"""
         self._test_channel_read()
         self.set_base_prompt()
-        self.set_terminal_width(command="stty columns 255", pattern="stty")
         self.disable_paging(command="term len 0")
-        # Clear the read buffer
+
+        # try ONEOS6 command first, differs from ONEOS5
+        self.set_terminal_width(command="screen-width 512")
+        output = self._test_channel_read()
+        if "error" in output.lower():
```

Comment:
Can you show what this looks like on the CLI for ONEOS5 when it fails (i.e. when that `error` line is in the output)?

Reply:
Below is the output of ONEOS5 + ONEOS6.  

Not sure, maybe it would be better to include some callback to check the OS flavor first and then run the appropriate commands for that version but in this case this slows down the initial connectivity a lot + it's only an added value if the OS version could be stored somewhere in the connection details so that it's always accessible.  But then again .. if there are too many differences between OS flavors then I guess it would make more sense to create a separate driver.

Should I include updates to snmp_autodect + ssh_autodetect for oneaccess_oneos as well?

Here's the output of **ONEOS5**:

DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: term len 0

DEBUG:netmiko:Pattern found: (term\ len\ 0) term len 0
DEBUG:netmiko:term len 0
DEBUG:netmiko:Exiting disable_paging
DEBUG:netmiko:write_channel: b'screen-width 512\r\n'
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: dops-lab-02#
DEBUG:netmiko:Pattern found: (dops\-lab\-02) 
dops-lab-02
DEBUG:netmiko:read_channel: screen-width 512
            ^
Error: Invalid command
dops-lab-02#
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:write_channel: b'stty columns 255\r\n'
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: stty columns 255

DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: dops-lab-02#
DEBUG:netmiko:Pattern found: (dops\-lab\-02) stty columns 255
dops-lab-02
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:Clear buffer detects data in the channel
DEBUG:netmiko:read_channel: 





Here's the output of **ONEOS6**:

DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: botlab01-02eve-01#
botlab01-02eve-01#
DEBUG:netmiko:Pattern found: (\#|>) 
botlab01-02eve-01#
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:Clear buffer detects data in the channel
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:[find_prompt()]: prompt is botlab01-02eve-01#
DEBUG:netmiko:In disable_paging
DEBUG:netmiko:Command: term len 0

DEBUG:netmiko:write_channel: b'term len 0\r\n'
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: term len 0
DEBUG:netmiko:Pattern found: (term\ len\ 0) term len 0
DEBUG:netmiko:term len 0
DEBUG:netmiko:Exiting disable_paging
DEBUG:netmiko:write_channel: b'screen-width 512\r\n'
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: